apis/apiextensions/v1alpha1/composition_types.go

```diff
@@ -77,6 +77,7 @@ type TypeReference struct {
 type ComposedTemplate struct {
 	// Base is the target resource that the patches will be applied on.
 	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:EmbeddedResource
```

Comment:
> An embedded resource is a value that has apiVersion, kind and metadata fields. They are validated implicitly according to the semantics of the currently running apiserver. It is not necessary to add any additional schema for these field, yet it is possible.

Will this allow `base`s without `metadata` field?

Reply:
It does. I think I left a comment about this in the commit message because I also read that it would require metadata.